Enable memfs tests that were disabled by mistake

ypatia opened the pull request:
In the past memfs was conditionally enabled based on a boostrap parameter, but was later changed to be enabled by default. However, some leftover enabled flag checking in unit tests was preventing memFS UTs from running.

+ some minor fixes to log status on every memfs error and a small addition to UTs.

---
TYPE: IMPROVEMENT
DESC: Enable memfs tests that were disabled by mistake

bekadavis9 reviewed (approved):
LGTM

Shelnutt2 commented:
I do not believe this change actually does anything different. @ypatia can you clarify what you are looking to do? Do you want a separate CI job that runs against memfs? As is this removes the checks for the filesystem support but doesn't do anything else?

ypatia commented:
> I do not believe this change actually does anything different. @ypatia can you clarify what you are looking to do? Do you want a separate CI job that runs against memfs? As is this removes the checks for the filesystem support but doesn't do anything else?

Sorry, I was probably not detailed enough in describing the issue. Back in the day when we introduced memFS it was conditionally compiled in based on a bootstrap option controlling a Cmake variable (as it was experimental), the same way it's done for HDFS, S3 etc.
Later, when it was properly tested/benchmarked we removed that option/variable in order to include it by default. However, we omitted to remove `supports_memfs_` from certain places in UT code so we ended up declaring the boolean but not intializing like we do for the others : https://github.com/TileDB-Inc/TileDB/blob/dev/test/src/unit-capi-vfs.cc#L114, and still using it (uninitialized) to decide if we are going to run the tests or not.
I think local uninitialized bool variables are undefined in C++, not false by default, but on the runs I did on my machine memFS test were not run because of this deprecated check.

By the way this diff coincidentally also fixes: sc11613

bekadavis9 commented:
The previous work item was done in February. The request [here](https://app.shortcut.com/tiledb-inc/story/4354/validate-enable-memfs) was to assume memFS is always enabled and make the appropriate changes. I think these changes @ypatia has made mostly support the theory that "memFS should always be enabled", the main change being the removal of boolean `supports_memfs_` and its conditional. It _may_ not be a totally _necessary_ change but I think it at least clarifies that memfs is _always_ enabled and doesn't leave room to user error, setting that bool to false.

ihnorton commented:
The merge build here failed:
```
2021-12-09T13:25:32.6473940Z [ 28%] Built target byteshuffle_filter
2021-12-09T13:25:32.6619950Z [ 28%] Building CXX object tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/compressors/bzip_compressor.cc.o
2021-12-09T13:25:32.6662580Z [ 28%] Built target bitshuffle_filter
2021-12-09T13:25:32.6803640Z [ 28%] Building CXX object tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/compressors/dd_compressor.cc.o
2021-12-09T13:25:34.7966140Z [ 28%] Building CXX object tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/compressors/gzip_compressor.cc.o
2021-12-09T13:25:35.6220700Z [ 31%] Building CXX object tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/compressors/lz4_compressor.cc.o
2021-12-09T13:25:35.6672990Z [ 31%] Building CXX object tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/compressors/rle_compressor.cc.o
2021-12-09T13:25:37.2166560Z [ 31%] Building CXX object tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/compressors/zstd_compressor.cc.o
2021-12-09T13:25:37.8846640Z [ 31%] Building CXX object tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/config/config.cc.o
2021-12-09T13:25:37.9083310Z [ 34%] Building CXX object tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/config/config_iter.cc.o
2021-12-09T13:25:40.3563920Z [ 34%] Building CXX object tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/crypto/crypto.cc.o
2021-12-09T13:25:40.4987810Z [ 34%] Building CXX object tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/crypto/encryption_key.cc.o
2021-12-09T13:25:42.9638820Z [ 34%] Building CXX object tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/crypto/encryption_key_validation.cc.o
2021-12-09T13:25:43.3971570Z [ 37%] Building CXX object tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/crypto/crypto_openssl.cc.o
2021-12-09T13:25:46.9938990Z [ 37%] Building CXX object tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/crypto/crypto_win32.cc.o
2021-12-09T13:25:46.9997720Z /Users/runner/work/TileDB/TileDB/tiledb/sm/crypto/crypto_openssl.cc:256:3: error: 'MD5' is deprecated [-Werror,-Wdeprecated-declarations]
2021-12-09T13:25:47.0094720Z   MD5(static_cast<const unsigned char*>(input),
2021-12-09T13:25:47.0118170Z   ^
2021-12-09T13:25:47.0137580Z /usr/local/opt/openssl/include/openssl/md5.h:52:1: note: 'MD5' has been explicitly marked deprecated here
2021-12-09T13:25:47.0208740Z OSSL_DEPRECATEDIN_3_0 unsigned char *MD5(const unsigned char *d, size_t n,
2021-12-09T13:25:47.0220720Z ^
2021-12-09T13:25:47.0243220Z /usr/local/opt/openssl/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
2021-12-09T13:25:47.0346340Z #   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
2021-12-09T13:25:47.0347160Z                                                 ^
2021-12-09T13:25:47.0379750Z /usr/local/opt/openssl/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
2021-12-09T13:25:47.0481650Z #     define OSSL_DEPRECATED(since) __attribute__((deprecated))
2021-12-09T13:25:47.0583250Z                                                    ^
2021-12-09T13:25:47.0752250Z [ 37%] Building CXX object tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/filesystem/azure.cc.o
2021-12-09T13:25:47.0777870Z 1 error generated.
2021-12-09T13:25:47.0833940Z make[5]: *** [tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/crypto/crypto_openssl.cc.o] Error 1
2021-12-09T13:25:47.0856410Z make[5]: *** Waiting for unfinished jobs....
2021-12-09T13:25:51.5577960Z make[4]: *** [tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/all] Error 2
2021-12-09T13:25:51.5580150Z make[3]: *** [all] Error 2
2021-12-09T13:25:51.5587070Z make[2]: *** [tiledb-prefix/src/tiledb-stamp/tiledb-build] Error 2
2021-12-09T13:25:51.5588680Z make[1]: *** [CMakeFiles/tiledb.dir/all] Error 2
2021-12-09T13:25:51.5591340Z make: *** [all] Error 2
2021-12-09T13:25:56.6249520Z ##[error]Process completed with exit code 2.
2021-12-09T13:25:56.6401800Z ##[group]Run source $GITHUB_WORKSPACE/scripts/ci/print_logs.sh
2021-12-09T13:25:56.6402690Z [36;1msource $GITHUB_WORKSPACE/scripts/ci/print_logs.sh[0m
```
